region/brinstar/green/Early Supers Room.json

```diff
@@ -357,6 +295,42 @@
         "It is also possible to clip into the gates by aiming down while in the spin jump."
       ]
     },
+    {
+      "link": [1, 2],
+      "name": "Leave Shinecharged (Speed Through)",
+      "notable": false,
+      "requires": [
+        {"obstaclesNotCleared": ["A"]},
+        {"canShineCharge": {
+          "usedTiles": 33,
+          "openEnd": 2
+        }}
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 110
+        }
+      }
+    },
+    {
+      "link": [1, 2],
+      "name": "Leave Shinecharged (Speed Through and Slide Off)",
+      "notable": false,
+      "requires": [
+        {"obstaclesNotCleared": ["A"]},
+        {"canShineCharge": {
+          "usedTiles": 33,
+          "openEnd": 2
+        }},
+        "canShinechargeMovementComplex"
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 145
+        }
+      },
+      "note": "Gain the shinecharge near the edge of the runway so that you slide off quickly, while shooting the door open."
```

Comment:
Slide off is more like an expert movement, then as a requirement a bit higher than that since this takes a room reset to retry.

Reply:
I see, do we need a new tech then? Sliding off like this is going to be useful in a lot of places for quick shinecharge movement. To me this feels much less difficult than Expert tech like Moat CWJ or Baby Skip for example. I was thinking Very Hard ("canShinechargeMovementComplex") would be fine

Reply:
Slide off doesn't actually feel that bad at mid run speeds so maybe it is ok in very hard.  I mostly remember it being difficult in speedruns because you also go full speed there.  In this case you have to run near full speed to beat the crumbles and shutters.  With the door opening shot, its pretty hard to do.

How about lowering frames remaining to either 140 for a middling number, or 135 for walking off the ledge while opening the door?

Reply:
Sure, lowered it to 140. I've also been thinking that we could add some leniency on the randomizer side, a number of frames to be subtracted from the 180 frames that you're initially assumed to have, maybe something like this:

Basic/Medium: 60 frames
Hard: 30 frames
Very Hard: 20 frames
Expert: 10 frames
Extreme: 5 frames
Insane: 0 frames

On the other hand, to make that work for in-room shinecharges, we would need to add a `framesRemaining` field to `canShinecharge`.

Reply:
It is feeling to me more important to add shinespark frames leniency.  It is another case where we would take off 10 or 20 frames as a way to add leniency, in the past.  And I take off closer to 5 now.

In room shinecharge strats should have the appropriate tech for the difficulty of the strat so wouldn't need leniency.